pkg/asset/installconfig/baremetal/host.go

```diff
@@ -0,0 +1,182 @@
+package baremetal
+
+import (
+	"strconv"
+
+	"github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+	"github.com/openshift/installer/pkg/types/baremetal"
+	"github.com/pkg/errors"
+	"gopkg.in/AlecAivazis/survey.v1"
+)
+
+// Host prompts the user for hardware details about a baremetal host.
+func Host() (*baremetal.Host, error) {
+	var host baremetal.Host
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Name",
+			},
+			Validate: survey.ComposeValidators(survey.Required),
+		},
+	}, &host.Name); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "BMC Address",
+				Help:    "The address for the BMC, e.g. ipmi://192.168.0.1",
+			},
+			Validate: survey.ComposeValidators(survey.Required, uriValidator),
+		},
+	}, &host.BMC.Address); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "BMC Username",
+			},
+			Validate: survey.ComposeValidators(survey.Required),
+		},
+	}, &host.BMC.Username); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Password{
+				Message: "BMC Password",
+			},
+			Validate: survey.ComposeValidators(survey.Required),
+		},
+	}, &host.BMC.Password); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Boot MAC Address",
+			},
+			Validate: survey.ComposeValidators(survey.Required, macValidator),
+		},
+	}, &host.BootMACAddress); err != nil {
+		return nil, err
+	}
+
+	// Boot mode
+	// Fixme: Wait until #3952 lands to uncomment this block
+	/*
+		var bootMode v1alpha1.BootMode
+		survey.AskOne(&survey.Select{
+			Message: "Select Boot Mode:",
+			Options: []string{"UEFI", "Legacy"},
+		}, &bootMode, nil)
+	*/
```

Comment:
We prefer not leaving commented code. So please remove it.

Reply:
secondly since boot mode has a default to UEFI, my preference would be to not prompt it for TUI.

Reply:
Agree. This was an oversight. We were going to remove this.